plan/relations.go

```diff
@@ -1136,6 +1570,62 @@ func (mr *MergeJoinRel) ToProtoPlanRel() *proto.PlanRel {
 	}
 }
 
+func (mr *MergeJoinRel) GetInputs() []Rel {
+	return []Rel{mr.left, mr.right}
+}
+
+func (mr *MergeJoinRel) Copy(newInputs ...Rel) (Rel, error) {
+	if len(newInputs) != 2 {
+		return nil, substraitgo.ErrInvalidInputCount
+	}
+	join := *mr
+	join.left = newInputs[0]
+	join.right = newInputs[1]
+	return &join, nil
+}
+
+func (mr *MergeJoinRel) CopyWithExpressionRewrite(rewriteFunc func(expr.Expression) (expr.Expression, error), newInputs ...Rel) (Rel, error) {
+	if len(newInputs) != 2 {
+		return nil, substraitgo.ErrInvalidInputCount
+	}
+	newExpr, err := rewriteFunc(mr.postJoinFilter)
+	if err != nil {
+		return nil, err
+	}
+	if newExpr == mr.postJoinFilter && areRelsEqual(newInputs, mr.GetInputs()) {
+		return nil, nil
+	}
+	merge := *mr
+	merge.left = newInputs[0]
+	merge.right = newInputs[1]
+	merge.postJoinFilter = newExpr
+	return &merge, nil
+}
+
+func areExpressionsEqual(exprs []expr.Expression, exprs2 []expr.Expression) bool {
```

Comment:
Equal might not be the best name here. Maybe adjust to something like Same or Identical with a doc describing behavior. same for are rels equal

Reply:
changed this to use slices.Equal()